src/Components/Web.JS/src/Platform/Circuits/CircuitManager.ts

```diff
@@ -131,20 +131,39 @@ export class CircuitManager implements DotNet.DotNetCallDispatcher {
     connection.on('JS.EndInvokeDotNet', this._dispatcher.endInvokeDotNetFromJS.bind(this._dispatcher));
     connection.on('JS.ReceiveByteArray', this._dispatcher.receiveByteArray.bind(this._dispatcher));
 
+    const failedStreamIds = new Set<number>();
+
     connection.on('JS.BeginTransmitStream', (streamId: number) => {
       const readableStream = new ReadableStream({
         start: (controller) => {
           connection.stream('SendDotNetStreamToJS', streamId).subscribe({
             next: (chunk: Uint8Array) => controller.enqueue(chunk),
-            complete: () => controller.close(),
-            error: (err) => controller.error(err),
+            complete: () => {
+              if (failedStreamIds.delete(streamId)) {
+                controller.error(new Error('Stream transmission failed. The stream may have been disposed.'));
+              } else {
+                controller.close();
+              }
+            },
```

Comment:
In what situation it can happen that `completed` gets called without `error` being called first? In other words, why do we need both error and complete handlers

Reply:
`error` only gets called when an exception gets thrown from the SignalR hub invocation. And I was trying to avoid that because SignalR treats that more as an error in the hub method logic, and we want to instead expose it via JS so it can get handled by the application.

So we want to support both kinds of errors (unhandled by the hub method _and_ handled by the hub method and "rethrown"), hence the two different locations where errors get reported.